File: deploy/android/src/haxor/context/PointGizmo.java
package haxor.context;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class PointGizmo extends haxor.context.Gizmo
{
	public    PointGizmo(haxe.lang.EmptyObject empty)
	{
		super(haxe.lang.EmptyObject.EMPTY);
	}
	
	
	public    PointGizmo()
	{
		super(haxe.lang.EmptyObject.EMPTY);
		haxor.context.PointGizmo.__hx_ctor_haxor_context_PointGizmo(this);
	}
	
	
	public static   void __hx_ctor_haxor_context_PointGizmo(haxor.context.PointGizmo __temp_me147)
	{
		haxor.context.Gizmo.__hx_ctor_haxor_context_Gizmo(__temp_me147, haxor.context.Gizmo.POINT, ( haxor.context.Gizmo.MAX_GIZMOS * 5 ));
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new haxor.context.PointGizmo(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new haxor.context.PointGizmo();
	}
	
	
	@Override public   void OnBuild()
	{
		haxor.graphics.mesh.Mesh3 m = new haxor.graphics.mesh.Mesh3(haxe.lang.Runtime.toString(null));
		m.set_name("$GizmoPointMesh");
		m.primitive = 0;
		double id = 0.0;
		haxor.io.FloatArray ia = new haxor.io.FloatArray(((int) (this.m_gizmo_count) ));
		haxor.io.FloatArray va = new haxor.io.FloatArray(((int) (( this.m_gizmo_count * 3 )) ));
		int va_k = 0;
		int ia_k = 0;
		{
			int _g1 = 0;
			int _g = this.m_gizmo_count;
			while (( _g1 < _g ))
			{
				int k = _g1++;
				va.Set(va_k++, 0.0);
				va.Set(va_k++, 0.0);
				va.Set(va_k++, 0.0);
				ia.Set(ia_k++, id);
				id += 1.0;
			}
			
		}
		
		m.Set("id", ia, 1);
		m.Set("vertex", va, 3);
		this.mesh = m;
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef851 = true;
			switch (field.hashCode())
			{
				case 312095663:
				{
					if (field.equals("OnBuild")) 
					{
						__temp_executeDef851 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("OnBuild"))) );
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef851) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
}

File: deploy/windows/src/haxor/context/PointGizmo.cpp
#include <hxcpp.h>

#ifndef INCLUDED_haxor_context_Gizmo
#include <haxor/context/Gizmo.h>
#endif
#ifndef INCLUDED_haxor_context_PointGizmo
#include <haxor/context/PointGizmo.h>
#endif
#ifndef INCLUDED_haxor_core_IDisposable
#include <haxor/core/IDisposable.h>
#endif
#ifndef INCLUDED_haxor_core_Resource
#include <haxor/core/Resource.h>
#endif
#ifndef INCLUDED_haxor_graphics_mesh_Mesh
#include <haxor/graphics/mesh/Mesh.h>
#endif
#ifndef INCLUDED_haxor_graphics_mesh_Mesh3
#include <haxor/graphics/mesh/Mesh3.h>
#endif
#ifndef INCLUDED_haxor_graphics_mesh_MeshAttrib
#include <haxor/graphics/mesh/MeshAttrib.h>
#endif
#ifndef INCLUDED_haxor_graphics_mesh_MeshLayout
#include <haxor/graphics/mesh/MeshLayout.h>
#endif
#ifndef INCLUDED_haxor_io_Buffer
#include <haxor/io/Buffer.h>
#endif
#ifndef INCLUDED_haxor_io_FloatArray
#include <haxor/io/FloatArray.h>
#endif
namespace haxor{
namespace context{

Void PointGizmo_obj::__construct()
{
HX_STACK_FRAME("haxor.context.PointGizmo","new",0xdc90fd67,"haxor.context.PointGizmo.new","haxor/context/GizmoContext.hx",647,0xa99c50bc)
HX_STACK_THIS(this)
{
	HX_STACK_LINE(647)
	super::__construct(::haxor::context::Gizmo_obj::POINT,(::haxor::context::Gizmo_obj::MAX_GIZMOS * (int)5));
}
;
	return null();
}

//PointGizmo_obj::~PointGizmo_obj() { }

Dynamic PointGizmo_obj::__CreateEmpty() { return  new PointGizmo_obj; }
hx::ObjectPtr< PointGizmo_obj > PointGizmo_obj::__new()
{  hx::ObjectPtr< PointGizmo_obj > result = new PointGizmo_obj();
	result->__construct();
	return result;}

Dynamic PointGizmo_obj::__Create(hx::DynamicArray inArgs)
{  hx::ObjectPtr< PointGizmo_obj > result = new PointGizmo_obj();
	result->__construct();
	return result;}

Void PointGizmo_obj::OnBuild( ){
{
		HX_STACK_FRAME("haxor.context.PointGizmo","OnBuild",0x6d1f5df6,"haxor.context.PointGizmo.OnBuild","haxor/context/GizmoContext.hx",653,0xa99c50bc)
		HX_STACK_THIS(this)
		HX_STACK_LINE(654)
		::haxor::graphics::mesh::Mesh3 m = ::haxor::graphics::mesh::Mesh3_obj::__new(null());		HX_STACK_VAR(m,"m");
		HX_STACK_LINE(655)
		m->set_name(HX_CSTRING("$GizmoPointMesh"));
		HX_STACK_LINE(656)
		m->primitive = (int)0;
		HX_STACK_LINE(657)
		Float id = 0.0;		HX_STACK_VAR(id,"id");
		HX_STACK_LINE(658)
		::haxor::io::FloatArray ia = ::haxor::io::FloatArray_obj::__new(this->m_gizmo_count);		HX_STACK_VAR(ia,"ia");
		HX_STACK_LINE(659)
		::haxor::io::FloatArray va = ::haxor::io::FloatArray_obj::__new((this->m_gizmo_count * (int)3));		HX_STACK_VAR(va,"va");
		HX_STACK_LINE(660)
		int va_k = (int)0;		HX_STACK_VAR(va_k,"va_k");
		HX_STACK_LINE(661)
		int ia_k = (int)0;		HX_STACK_VAR(ia_k,"ia_k");
		HX_STACK_LINE(662)
		{
			HX_STACK_LINE(662)
			int _g1 = (int)0;		HX_STACK_VAR(_g1,"_g1");
			HX_STACK_LINE(662)
			int _g = this->m_gizmo_count;		HX_STACK_VAR(_g,"_g");
			HX_STACK_LINE(662)
			while((true)){
				HX_STACK_LINE(662)
				if ((!(((_g1 < _g))))){
					HX_STACK_LINE(662)
					break;
				}
				HX_STACK_LINE(662)
				int k = (_g1)++;		HX_STACK_VAR(k,"k");
				HX_STACK_LINE(664)
				int _g2 = (va_k)++;		HX_STACK_VAR(_g2,"_g2");
				HX_STACK_LINE(664)
				va->Set(_g2,0.0);
				HX_STACK_LINE(664)
				int _g11 = (va_k)++;		HX_STACK_VAR(_g11,"_g11");
				HX_STACK_LINE(664)
				va->Set(_g11,0.0);
				HX_STACK_LINE(664)
				int _g21 = (va_k)++;		HX_STACK_VAR(_g21,"_g21");
				HX_STACK_LINE(664)
				va->Set(_g21,0.0);
				HX_STACK_LINE(664)
				int _g3 = (ia_k)++;		HX_STACK_VAR(_g3,"_g3");
				HX_STACK_LINE(664)
				ia->Set(_g3,id);
				HX_STACK_LINE(665)
				hx::AddEq(id,1.0);
			}
		}
		HX_STACK_LINE(667)
		m->Set(HX_CSTRING("id"),ia,(int)1);
		HX_STACK_LINE(668)
		m->Set(HX_CSTRING("vertex"),va,(int)3);
		HX_STACK_LINE(669)
		this->mesh = m;
	}
return null();
}



PointGizmo_obj::PointGizmo_obj()
{
}

Dynamic PointGizmo_obj::__Field(const ::String &inName,bool inCallProp)
{
	switch(inName.length) {
	case 7:
		if (HX_FIELD_EQ(inName,"OnBuild") ) { return OnBuild_dyn(); }
	}
	return super::__Field(inName,inCallProp);
}

Dynamic PointGizmo_obj::__SetField(const ::String &inName,const Dynamic &inValue,bool inCallProp)
{
	return super::__SetField(inName,inValue,inCallProp);
}

void PointGizmo_obj::__GetFields(Array< ::String> &outFields)
{
	super::__GetFields(outFields);
};

static ::String sStaticFields[] = {
	String(null()) };

#if HXCPP_SCRIPTABLE
static hx::StorageInfo *sMemberStorageInfo = 0;
#endif

static ::String sMemberFields[] = {
	HX_CSTRING("OnBuild"),
	String(null()) };

static void sMarkStatics(HX_MARK_PARAMS) {
	HX_MARK_MEMBER_NAME(PointGizmo_obj::__mClass,"__mClass");
};

#ifdef HXCPP_VISIT_ALLOCS
static void sVisitStatics(HX_VISIT_PARAMS) {
	HX_VISIT_MEMBER_NAME(PointGizmo_obj::__mClass,"__mClass");
};

#endif

Class PointGizmo_obj::__mClass;

void PointGizmo_obj::__register()
{
	hx::Static(__mClass) = hx::RegisterClass(HX_CSTRING("haxor.context.PointGizmo"), hx::TCanCast< PointGizmo_obj> ,sStaticFields,sMemberFields,
	&__CreateEmpty, &__Create,
	&super::__SGetClass(), 0, sMarkStatics
#ifdef HXCPP_VISIT_ALLOCS
    , sVisitStatics
#endif
#ifdef HXCPP_SCRIPTABLE
    , sMemberStorageInfo
#endif
);
}

void PointGizmo_obj::__boot()
{
}

} // end namespace haxor
} // end namespace context
